### DISCUSSION TITLE
Why converting to float64 instead of float32 in case of masked=True

### DISCUSSION THREAD

**Discussion by remi-braun** [corteva/rioxarray#302] [Q&A]
Hello, me again 😅 

I was wondering why the data is converted to float64 and not float32 when a dataset is read with `masked=True` (and original data is uin16 for example)
Isn't it dangerous wrt memory consumption ?
(I am not totally fluent in xarray, so it might be an obvious reason that I don't see 😓)

**Comment by remi-braun** [comment]
And if I have understood everything correctly, the conversion to float is forced because of the use of nan, is it right ?

**Reply by snowman2** [reply]
> the conversion to float is forced because of the use of nan, is it right ?

Correct

**Comment by snowman2** [comment]
It currently matches the behavior of`xarray.open_dataset` and is the safest datatype to convert to.

> Isn't it dangerous wrt memory consumption ?

It hasn't been optimized for memory at this point. PRs are welcome :smile:

**Reply by remi-braun** [reply]
If even `xarray` converts everything to float64, they may have thought this through. 
Maybe this makes no problem as long you don't call `load()` on the array...

Thanks for your very quick replies !

**Reply by snowman2** [reply]
Hmmm, maybe I was wrong on that one. Related: https://github.com/pydata/xarray/issues/2304

**Reply by snowman2** [reply]
Looks like it has been updated: https://github.com/pydata/xarray/blob/e0358e586079c12525ce60c4a51b591dc280713b/xarray/core/dtypes.py#L45-L78

**Reply by snowman2** [reply]
See #304
